src/DynamoCoreWpf/ViewModels/PackageManager/PublishPackageViewModel.cs

```diff
@@ -540,6 +541,56 @@ public PublishPackageViewModel( DynamoViewModel dynamoViewModel ) : this()
             this.dynamoViewModel = dynamoViewModel;
         }
 
+        private void clearAllEntries()
+        {
+            // this function clears all the entries of the publish package dialog
+            this.Name = "";
+            this.RepositoryUrl = "";
+            this.SiteUrl = "";
+            this.License = "";
+            this.Keywords = "";
+            this.Description = "";
+            this.Group = "";
+            this.clearFullVersion();
+
+            this.Package = null;
+            this.Uploading = false;
+            this.UploadHandle = null;
+            this.IsNewVersion = false;
+            this.AdditionalFiles = new ObservableCollection<string>();
+            this.ErrorString = "";
+            this.MoreExpanded = false;
+            this.UploadState = PackageUploadHandle.State.Ready;
+
+            this.SubmitCommand.RaiseCanExecuteChanged();
```

Comment:
Notice `MajorVersion` setter will call `SubmitCommand.RaiseCanExecuteChanged()`, I think probably this function will be invoked in the other setters as well.